File: tipple-service/src/main/java/de/tipple/model/Configuration.java
package de.tipple.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;
@Document
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Configuration {
  @Id
  private String id;
  private String userId;
  private LocalDateTime date;
  private boolean share;
  private boolean cart;
  @JsonInclude(JsonInclude.Include.NON_NULL)
  private List<Ingredient> ingredients;
  public Configuration() {
  }

  public Configuration(String id, String userId, LocalDateTime date,
                       boolean share, boolean chart, List<Ingredient> ingredientList) {
    this.id = id;
    this.userId = userId;
    this.date = date;
    this.share = share;
    this.cart = chart;
    this.ingredients = ingredientList;
  }
  public Configuration(Configuration config) {
    this(config.id, config.userId, config.getDate(),
        config.isShare(), config.isCart(), config.getIngredients());
  }

  public boolean isCart() {
    return cart;
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getUserId() {
    return userId;
  }

  public boolean isShare() {
    return share;
  }

  public List<Ingredient> getIngredients() {
    return ingredients;
  }

  public LocalDateTime getDate() {
    return date;
  }
}

File: lib/configuration/configuration.dart
import 'package:tipple_app/ingredient/ingredient.dart';

class Configuration {
  final String id;
  final String userId;
  String date;
  bool share;
  bool cart;
  List<Ingredient> ingredient;
  Configuration({
      this.id,
      this.userId,
      this.date,
      this.share,
      this.cart,
      this.ingredient});

  factory Configuration.fromJson(Map<String, dynamic> json) {
    return Configuration(
        id: json['id'],
        userId: json['userId'],
        date: json['date'],
        share: json['share'],
        cart: json['cart'],
        ingredient: (json['ingredients'] as List)
            .map((p) => Ingredient.fromJson(p))
            .toList());
  }
}
